include/powsybl/iidm/LoadingLimitsAdder.hxx

```diff
@@ -0,0 +1,176 @@
+
+namespace iidm {
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>::TemporaryLimitAdder::TemporaryLimitAdder(OperationalLimitsHolder& owner, LoadingLimitsAdder<L, A>& parent) :
+    m_owner(owner),
+    m_parent(parent) {
+}
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::endTemporaryLimit() {
+    if (std::isnan(m_value)) {
+        throw ValidationException(m_owner, "temporary limit value is not set");
+    }
+    if (m_value <= 0) {
+        throw ValidationException(m_owner, "temporary limit value must be > 0");
+    }
+    if (!m_acceptableDuration) {
+        throw ValidationException(m_owner, "acceptable duration is not set");
+    }
+    checkAndGetUniqueName();
+    return m_parent.addTemporaryLimit(m_name, m_value, *m_acceptableDuration, m_fictitious);
+}
+
+template <typename L, typename A>
+void LoadingLimitsAdder<L, A>::TemporaryLimitAdder::checkAndGetUniqueName() {
+    if (m_name.empty()) {
+        throw ValidationException(m_owner, "name is not set");
+    }
+    if (m_ensureNameUnicity) {
+        unsigned long i = 0UL;
+        std::string uniqueName = m_name;
+        while (i < std::numeric_limits<unsigned long>::max() && m_parent.nameExists(uniqueName)) {
+            uniqueName = m_name + stdcxx::format("#%1%", i);
+            i++;
+        }
+        m_name = uniqueName;
+    }
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::ensureNameUnicity() {
+    m_ensureNameUnicity = true;
+    return *this;
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::setAcceptableDuration(unsigned long duration) {
+    m_acceptableDuration = duration;
+    return *this;
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::setFictitious(bool fictitious) {
+    m_fictitious = fictitious;
+    return *this;
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::setName(const std::string& name) {
+    m_name = name;
+    return *this;
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::setValue(double value) {
+    m_value = value;
+    return *this;
+}
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>::LoadingLimitsAdder(OperationalLimitsHolder& owner) :
+    m_owner(owner) {
+}
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>& LoadingLimitsAdder<L, A>::addTemporaryLimit(const std::string& name, double value, unsigned long acceptableDuration, bool fictitious) {
+    m_temporaryLimits.emplace(acceptableDuration, CurrentLimits::TemporaryLimit(name, value, acceptableDuration, fictitious));
```

Comment:
This is really stange: `TermporaryLimits` should not be prefixed by `CurrentLimits` as CurrentLimits is a subclass of `LoadingLimits`. Replace with `LoadingLimits` ?